app/models/piece.rb

```diff
@@ -18,12 +17,47 @@ def move!(x, y)
     true
   end
 
+  # Return false if coordinates is taken but belongs to friendly piece
+  # otherwise return true
+  def capturable?(x, y)
+    return false unless position_taken?(x, y) && !friendly_piece?(x, y)
+    true
+  end
+
+  # if piece is capturable change piece's attributes
+  def capture!(x, y)
+    return false unless capturable?(x, y)
+    move!(x, y)
+    target_piece(x, y).update_attributes!(
+      x_position: nil,
+      y_position: nil,
+      captured: true
+    )
+  end
```

Comment:
Instead of calling move in capture, I would integrate capturable/capture into move.  Here's the flow:

- Player wants to move his rook to 3,5.
- We call rook.move!(3,5).
- Move checks if a rook can make that move, otherwise reject move
- Move checks if the path is not obstructed between origin and destination, otherwise reject move
- Move checks that if there is a piece in the tile, that it is hostile, otherwise reject move
- Update coordinates of mover, and capture victim if there was one